usr/bin/revpi-btuart

```diff
@@ -4,5 +4,15 @@
 HCIATTACH=/usr/bin/hciattach
 TTYDEVNAME="$1"
 
+# RevPi Flat: The BT_REG_ON signal is connected to the GPIO11 of the expander.
+GPIOCHIP=gpiochip3
+BT_REG_ON_GPIO=11
+
+# Reset the bt module in RevPi Flat
+gpioset ${GPIOCHIP} ${BT_REG_ON_GPIO}=0
```

Comment:
This script is "wanted" by "_ACTION=="add", SUBSYSTEM=="tty" DEVPATH=="*/usb1/1-1/1-1.5/1-1.5:1.0/*", TAG+="systemd",.._.". 
Is this for sure after the GPIOCHIP be available?

Reply:
The UART is not affected by this in any way. Only the bt module will be resetted. And as I understand it, the first access to the bt module will be with the hciattach. Or am I wrong?

Reply:
I understand the sequence of initialization and script being executed:
sequence 1: CPU -> i2c -> gpiochip3(expander)
sequence 2: CPU -> usb -> hub -> uart -> bt module --(by udev)--> revpi-btuart
And in the revpi-btuart script, it is assume that the expander is ready. 
But I am not sure whether the sequence 1 is always before sequence 2, I just ask, maybe something I don't know yet.

Reply:
Ok so your concern is that the expander might not be read when we start the bt initialization?

Reply:
yes. Btw the bt initialization is not started by us, but by udev, isn't it?

Reply:
I can assure you, that the i2c gpio expander is ready a long time before the bt initialization starts. I checked it with a logic analyzer and it is several seconds. The wifi module uses the same gpio expander. The reset of the wifi module is done from the mmc/sdio subsystem pretty early in the boot process. So I don't think we have any problem here.
There is no "us" in the system. I'm not sitting in any flat device and pulling gpios :-P
We (or much more accurately: you) provide the udev rule and the revpi-btuart script. So we are doing this. But yes it is triggered by a udev rule.

Are you pointing out, that there might be a race condition between the revpi-btusrt and the revpi-bthelper scripts? As bot scripts are called for systemd units, which are only added after a certain event occurred this is unlikely. The revpi-bthelper waits for the hci device. The hci device will be available after the hciattach. At least this is my understanding.

Reply:
it is clear, thanks a lot :)